packages/trie/test/proof.spec.ts

```diff
@@ -154,3 +155,254 @@ tape('simple merkle proofs generation and verification', function (tester) {
+
+    const proof = await trie.createRangeProof(lKey, rKey)
+
+    // This should report a part of the trie, somewhere between 1 and 9 elements
+
+    const vLen = proof.values.length
+    const kLen = proof.keys.length
+
+    t.ok(vLen === kLen)
+    t.ok(vLen > 1 && vLen < 9)
+
+    await proverTrie.verifyRangeProof(
+      trie.root(),
+      lKey,
+      rKey,
+      proof.keys,
+      proof.values,
+      proof.proof
+    )
+
+    t.end()
+  })
+
+  it('creates zero key/value proof', async (t) => {
+    // In this case, there are no key/values between the left and the right key
+    const trie = new Trie({
+      useKeyHashing: true,
+    })
+
+    const proverTrie = new Trie()
+
+    await trie.put(Buffer.from('1000', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1100', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1110', 'hex'), Buffer.from('a'))
+
+    await trie.put(Buffer.from('2000', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2200', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2220', 'hex'), Buffer.from('b'))
+
+    await trie.put(Buffer.from('3000', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3300', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3330', 'hex'), Buffer.from('c'))
+
+    const lKey = Buffer.from('EF'.repeat(32), 'hex')
+    const rKey = Buffer.from('FF'.repeat(32), 'hex')
+
+    const proof = await trie.createRangeProof(lKey, rKey)
+
+    await proverTrie.verifyRangeProof(
+      trie.root(),
+      lKey,
+      rKey,
+      proof.keys,
+      proof.values,
+      proof.proof
+    )
+
+    t.end()
+  })
+
+  it('creates all elements proof', async (t) => {
+    // In this case, report all values of the trie
+    const trie = new Trie({
+      useKeyHashing: true,
+    })
+
+    const proverTrie = new Trie()
+
+    await trie.put(Buffer.from('1000', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1100', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1110', 'hex'), Buffer.from('a'))
+
+    await trie.put(Buffer.from('2000', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2200', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2220', 'hex'), Buffer.from('b'))
+
+    await trie.put(Buffer.from('3000', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3300', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3330', 'hex'), Buffer.from('c'))
+
+    const lKey = Buffer.from('00'.repeat(32), 'hex')
+    const rKey = Buffer.from('FF'.repeat(32), 'hex')
+
+    const proof = await trie.createRangeProof(lKey, rKey)
+
+    // We do NOT need the multiproof here
+    // However, if we want to check this, we have to:
+    // (1) check if there is no item left of lKey
+    // (2) check if there is no item right of rKey
+    // This is trivial in a flat DB, but this is not yet supported
+    // Therefore, probably do not implement this.
+    t.ok(proof.proof.length === 0)
```

Comment:
I get an error here:
```
# creates all elements proof
not ok 23 should be truthy
  ---
    operator: ok
    expected: true
    actual:   false
    at: Test.<anonymous> (/ethjs/ethereumjs-monorepo/packages/trie/test/proof.spec.ts:335:7)
    stack: |-
      Error: should be truthy
          at Test.assert [as _assert] (/ethjs/ethereumjs-monorepo/node_modules/tape/lib/test.js:314:54)
          at Test.bound [as _assert] (/ethjs/ethereumjs-monorepo/node_modules/tape/lib/test.js:99:32)
          at Test.assert (/ethjs/ethereumjs-monorepo/node_modules/tape/lib/test.js:433:10)
          at Test.bound [as ok] (/ethjs/ethereumjs-monorepo/node_modules/tape/lib/test.js:99:32)
          at Test.<anonymous> (/ethjs/ethereumjs-monorepo/packages/trie/test/proof.spec.ts:335:7)
          at processTicksAndRejections (node:internal/process/task_queues:95:5)
          at runNextTicks (node:internal/process/task_queues:64:3)
          at processImmediate (node:internal/timers:442:9)
  ...
  ```
  Is there a reason empty proof is expected here?

Reply:
We do not need these proof items, because /all/ key/values are reported in the key/values :) (So to verify it is trivial, just dump in all key/values and check the root)